examples/views/paginator.py

```diff
@@ -1,49 +1,253 @@
         view.add_item(
             discord.ui.Select(
                 placeholder="Test Select Menu, Does Nothing",
-                options=[discord.SelectOption(label="Example Option", value="Example Value", description="This menu does nothing!")],
+                options=[
+                    discord.SelectOption(
+                        label="Example Option",
+                        value="Example Value",
+                        description="This menu does nothing!",
+                    )
+                ],
+            )
+        )
+        paginator = pages.Paginator(pages=self.get_pages(), custom_view=view)
+        await paginator.respond(ctx.interaction, ephemeral=False)
+
+    @pagetest.command(name="groups")
+    async def pagetest_groups(self, ctx: discord.ApplicationContext):
+        """Demonstrates using page groups to switch between different sets of pages."""
+        page_buttons = [
+            pages.PaginatorButton(
+                "first", label="<<-", style=discord.ButtonStyle.green
+            ),
+            pages.PaginatorButton("prev", label="<-", style=discord.ButtonStyle.green),
+            pages.PaginatorButton(
+                "page_indicator", style=discord.ButtonStyle.gray, disabled=True
+            ),
+            pages.PaginatorButton("next", label="->", style=discord.ButtonStyle.green),
+            pages.PaginatorButton("last", label="->>", style=discord.ButtonStyle.green),
+        ]
+        view = discord.ui.View()
+        view.add_item(discord.ui.Button(label="Test Button, Does Nothing", row=2))
+        view.add_item(
+            discord.ui.Select(
+                placeholder="Test Select Menu, Does Nothing",
+                options=[
+                    discord.SelectOption(
+                        label="Example Option",
+                        value="Example Value",
+                        description="This menu does nothing!",
+                    )
+                ],
             )
         )
-        paginator = pages.Paginator(pages=self.get_pages(), show_disabled=False, show_indicator=True, custom_view=view)
-        await paginator.send(ctx, ephemeral=False)
+        page_groups = [
+            pages.PageGroup(
+                pages=self.get_pages(),
+                label="Main Page Group",
+                description="Main Pages for Main Things",
+            ),
+            pages.PageGroup(
+                pages=[
+                    "Second Set of Pages, Page 1",
+                    "Second Set of Pages, Page 2",
+                    "Look, it's group 2, page 3!",
+                ],
+                label="Second Page Group",
+                description="Secondary Pages for Secondary Things",
+                custom_buttons=page_buttons,
+                use_default_buttons=False,
+                custom_view=view,
+            ),
+        ]
+        paginator = pages.Paginator(pages=page_groups, show_menu=True)
```

Comment:
Is it wanted that the dropdown always resets back to `Select Page Group` instead of staying at the selected option?

Reply:
This is the default behavior of selects and ext.pages doesn't currently update the select post-selection to change the displayed page group. Will probably add it as an option for 2.1 though if there's enough demand.